app/services/agent_status_updater.rb

```diff
@@ -22,7 +22,8 @@ def update_statuses(previous_statuses, new_statuses)
 
 	def close_last_open_status(agent)
 		open_statuses = AgentStatus.where(agent_id: agent.agent_id, open: true)
-		open_statuses.each { |status| status.open = false; status.closed = Time.now; status.save } #There should only ever be one open status at a time for each agent, looping through just in case
+		# There should only ever be one open status at a time for each agent, looping through just in case
+		open_statuses.each { |status| status.open = false; status.closed = Time.now; status.save } 
```

Comment:
Kinda unrelated to this pull request but ActiveRecord `update_all` could be used instead of looping. I think it would be clearer (and possibly more efficient as it should do only 1 query)